libs/s25main/GameCommand.cpp

```diff
@@ -9,7 +9,12 @@
 
 namespace gc {
 
-GameCommandPtr GameCommand::Deserialize(Serializer& ser)
+unsigned Deserializer::getCurrentVersion()
+{
+    return 0;
```

Comment:
can we have an explanation here what the value does?

Reply:
Do we need that here again? The documentation is at [`VersionedDeserializer`](https://github.com/Return-To-The-Roots/libutil/compare/98d6ab5b21bf7f35a2d5ee76d6ea43c7d416f8a1...a40c2337e87bec26ae495d7186e002d6cb0bb8bf#diff-310d747bd13159787f152c327cee22ff62a4c0ce15ee504c638c36846242cde1R14-R21)

I'm also not sure what to write here. It would be something like "Current version of the game data", but most of that is already in the method name. Any suggestion?

BTW: One use example is in a branch I created on top of the wine addon PR: https://github.com/Flamefire/s25client/commit/214173bcf4ca6ee90568024bef06afec5af7d0fa

Reply:
> The documentation is at [VersionedDeserializer](https://github.com/Return-To-The-Roots/libutil/compare/98d6ab5b21bf7f35a2d5ee76d6ea43c7d416f8a1...a40c2337e87bec26ae495d7186e002d6cb0bb8bf#diff-310d747bd13159787f152c327cee22ff62a4c0ce15ee504c638c36846242cde1R14-R21)
maybe exactly this. That should be enough.

is 0 now the version? if so, please make it a named constant, so its even more clear what it is.

Reply:
I added that as a docstring.

This already is a named constant. But I see that a function could do more than just returning the constant. I changed it to a static member constant to make that clear. The original intention was that a function returning a value leaves no doubt that its value cannot be modified, as opposed to a member variable where it was forgotten to make it constant.